WebXR Layers idl

mwyrzykowski opened the pull request:
#### 6920df14d2fb92eb75136dc718a35a74e3e05ff2
<pre>
WebXR Layers idl
<a href="https://bugs.webkit.org/show_bug.cgi?id=277168">https://bugs.webkit.org/show_bug.cgi?id=277168</a>
<a href="https://rdar.apple.com/132594643">rdar://132594643</a>

Reviewed by Tadeu Zagallo.

Patch from Dan. Add idl files for WebXR Layers.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRRenderState+Layers.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/WebXRRenderState.h:
(WebCore::WebXRRenderState::layers const):
* Source/WebCore/Modules/webxr/XRCompositionLayer.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCompositionLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCompositionLayer::layout const):
(WebCore::XRCompositionLayer::blendTextureSourceAlpha const):
(WebCore::XRCompositionLayer::setBlendTextureSourceAlpha):
(WebCore::XRCompositionLayer::forceMonoPresentation const):
(WebCore::XRCompositionLayer::setForceMonoPresentation):
(WebCore::XRCompositionLayer::opacity const):
(WebCore::XRCompositionLayer::setOpacity):
(WebCore::XRCompositionLayer::mipLevels const):
(WebCore::XRCompositionLayer::quality const):
(WebCore::XRCompositionLayer::setQuality):
(WebCore::XRCompositionLayer::needsRedraw const):
(WebCore::XRCompositionLayer::destroy):
* Source/WebCore/Modules/webxr/XRCompositionLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCubeLayer::space const):
(WebCore::XRCubeLayer::setSpace):
(WebCore::XRCubeLayer::orientation const):
(WebCore::XRCubeLayer::setOrientation):
* Source/WebCore/Modules/webxr/XRCubeLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCylinderLayer::space const):
(WebCore::XRCylinderLayer::setSpace):
(WebCore::XRCylinderLayer::transform const):
(WebCore::XRCylinderLayer::setTransform):
(WebCore::XRCylinderLayer::radius const):
(WebCore::XRCylinderLayer::setRadius):
(WebCore::XRCylinderLayer::centralAngle const):
(WebCore::XRCylinderLayer::setCentralAngle):
(WebCore::XRCylinderLayer::aspectRatio const):
(WebCore::XRCylinderLayer::setAspectRatio):
* Source/WebCore/Modules/webxr/XRCylinderLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XREquirectLayer::space const):
(WebCore::XREquirectLayer::setSpace):
(WebCore::XREquirectLayer::transform const):
(WebCore::XREquirectLayer::setTransform):
(WebCore::XREquirectLayer::radius const):
(WebCore::XREquirectLayer::setRadius):
(WebCore::XREquirectLayer::centralHorizontalAngle const):
(WebCore::XREquirectLayer::setCentralHorizontalAngle):
(WebCore::XREquirectLayer::upperVerticalAngle const):
(WebCore::XREquirectLayer::setUpperVerticalAngle):
(WebCore::XREquirectLayer::lowerVerticalAngle const):
(WebCore::XREquirectLayer::setLowerVerticalAngle):
* Source/WebCore/Modules/webxr/XREquirectLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEvent.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRLayerEvent::create):
(WebCore::XRLayerEvent::XRLayerEvent):
(WebCore::XRLayerEvent::eventInterfaceType const):
(WebCore::XRLayerEvent::layer const):
* Source/WebCore/Modules/webxr/XRLayerEvent.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEvent.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEventInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerLayout.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerLayout.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerQuality.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerQuality.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayer.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRProjectionLayer::startFrame):
(WebCore::XRProjectionLayer::endFrame):
* Source/WebCore/Modules/webxr/XRProjectionLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRProjectionLayer::textureWidth const):
(WebCore::XRProjectionLayer::textureHeight const):
(WebCore::XRProjectionLayer::textureArrayLength const):
(WebCore::XRProjectionLayer::ignoreDepthValues const):
(WebCore::XRProjectionLayer::fixedFoveation const):
(WebCore::XRProjectionLayer::setFixedFoveation):
(WebCore::XRProjectionLayer::deltaPose const):
(WebCore::XRProjectionLayer::setDeltaPose):
* Source/WebCore/Modules/webxr/XRProjectionLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRQuadLayer::space const):
(WebCore::XRQuadLayer::setSpace):
(WebCore::XRQuadLayer::transform const):
(WebCore::XRQuadLayer::setTransform):
(WebCore::XRQuadLayer::width const):
(WebCore::XRQuadLayer::setWidth):
(WebCore::XRQuadLayer::height const):
(WebCore::XRQuadLayer::setHeight):
* Source/WebCore/Modules/webxr/XRQuadLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRSubImage.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRSubImage::viewport const):
* Source/WebCore/Modules/webxr/XRSubImage.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRTextureType.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRTextureType.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLBinding.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRWebGLBinding::create):
* Source/WebCore/Modules/webxr/XRWebGLBinding.h: Added.
(WebCore::XRWebGLBinding::nativeProjectionScaleFactor const):
(WebCore::XRWebGLBinding::usesDepthValues const):
(WebCore::XRWebGLBinding::createProjectionLayer):
(WebCore::XRWebGLBinding::createQuadLayer):
(WebCore::XRWebGLBinding::createCylinderLayer):
(WebCore::XRWebGLBinding::createEquirectLayer):
(WebCore::XRWebGLBinding::createCubeLayer):
(WebCore::XRWebGLBinding::getSubImage):
(WebCore::XRWebGLBinding::getViewSubImage):
* Source/WebCore/Modules/webxr/XRWebGLBinding.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLSubImage.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLSubImage.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRWebGLSubImage::colorTexture const):
(WebCore::XRWebGLSubImage::depthStencilTexture const):
(WebCore::XRWebGLSubImage::motionVectorTexture const):
(WebCore::XRWebGLSubImage::imageIndex const):
(WebCore::XRWebGLSubImage::colorTextureWidth const):
(WebCore::XRWebGLSubImage::colorTextureHeight const):
(WebCore::XRWebGLSubImage::depthStencilTextureWidth const):
(WebCore::XRWebGLSubImage::depthStencilTextureHeight const):
(WebCore::XRWebGLSubImage::motionVectorTextureWidth const):
(WebCore::XRWebGLSubImage::motionVectorTextureHeight const):
* Source/WebCore/Modules/webxr/XRWebGLSubImage.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventInterfaces.in:
* Source/WebCore/dom/EventNames.json:
* Source/WebCore/html/HTMLAttributeNames.in:

Canonical link: <a href="https://commits.webkit.org/282031@main">https://commits.webkit.org/282031@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/02d763c613697687da53e93996270178adb3b3a8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61762 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41116 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14354 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65741 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12307 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/63881 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48802 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12578 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49812 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8546 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64831 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38194 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53499 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30644 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34850 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10729 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11238 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/54857 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56656 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11032 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67469 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/61003 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5705 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10788 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57189 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5730 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53446 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57448 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13752 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4693 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/82765 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36916 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14497 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38000 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39096 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37745 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebCore/Modules/webxr/XRCompositionLayer.cpp:
```diff
@@ -0,0 +1,41 @@
+#include "config.h"
+#include "XRCompositionLayer.h"
+
+#if ENABLE(WEBXR)
```
Does not match the .h

kkinnunen-apple commented on Source/WebCore/Modules/webxr/WebXRRenderState+Layers.idl:
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2023 Apple, Inc. All rights reserved.
```
year

kkinnunen-apple commented on Source/WebCore/Modules/webxr/XRCompositionLayer.cpp:
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2020 Igalia S.L. All rights reserved.
```
year, company ?

kkinnunen-apple commented on Source/WebCore/Modules/webxr/XRCompositionLayer.h:
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2023 Apple, Inc. All rights reserved.
```
Year

mwyrzykowski committed:
WebXR Layers idl
https://bugs.webkit.org/show_bug.cgi?id=277168
rdar://132594643

Reviewed by Tadeu Zagallo.

Patch from Dan. Add idl files for WebXR Layers.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXRRenderState+Layers.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/WebXRRenderState.h:
(WebCore::WebXRRenderState::layers const):
* Source/WebCore/Modules/webxr/XRCompositionLayer.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCompositionLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCompositionLayer::layout const):
(WebCore::XRCompositionLayer::blendTextureSourceAlpha const):
(WebCore::XRCompositionLayer::setBlendTextureSourceAlpha):
(WebCore::XRCompositionLayer::forceMonoPresentation const):
(WebCore::XRCompositionLayer::setForceMonoPresentation):
(WebCore::XRCompositionLayer::opacity const):
(WebCore::XRCompositionLayer::setOpacity):
(WebCore::XRCompositionLayer::mipLevels const):
(WebCore::XRCompositionLayer::quality const):
(WebCore::XRCompositionLayer::setQuality):
(WebCore::XRCompositionLayer::needsRedraw const):
(WebCore::XRCompositionLayer::destroy):
* Source/WebCore/Modules/webxr/XRCompositionLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCubeLayer::space const):
(WebCore::XRCubeLayer::setSpace):
(WebCore::XRCubeLayer::orientation const):
(WebCore::XRCubeLayer::setOrientation):
* Source/WebCore/Modules/webxr/XRCubeLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCubeLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRCylinderLayer::space const):
(WebCore::XRCylinderLayer::setSpace):
(WebCore::XRCylinderLayer::transform const):
(WebCore::XRCylinderLayer::setTransform):
(WebCore::XRCylinderLayer::radius const):
(WebCore::XRCylinderLayer::setRadius):
(WebCore::XRCylinderLayer::centralAngle const):
(WebCore::XRCylinderLayer::setCentralAngle):
(WebCore::XRCylinderLayer::aspectRatio const):
(WebCore::XRCylinderLayer::setAspectRatio):
* Source/WebCore/Modules/webxr/XRCylinderLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRCylinderLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XREquirectLayer::space const):
(WebCore::XREquirectLayer::setSpace):
(WebCore::XREquirectLayer::transform const):
(WebCore::XREquirectLayer::setTransform):
(WebCore::XREquirectLayer::radius const):
(WebCore::XREquirectLayer::setRadius):
(WebCore::XREquirectLayer::centralHorizontalAngle const):
(WebCore::XREquirectLayer::setCentralHorizontalAngle):
(WebCore::XREquirectLayer::upperVerticalAngle const):
(WebCore::XREquirectLayer::setUpperVerticalAngle):
(WebCore::XREquirectLayer::lowerVerticalAngle const):
(WebCore::XREquirectLayer::setLowerVerticalAngle):
* Source/WebCore/Modules/webxr/XREquirectLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XREquirectLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEvent.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRLayerEvent::create):
(WebCore::XRLayerEvent::XRLayerEvent):
(WebCore::XRLayerEvent::eventInterfaceType const):
(WebCore::XRLayerEvent::layer const):
* Source/WebCore/Modules/webxr/XRLayerEvent.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEvent.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerEventInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerLayout.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerLayout.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerQuality.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRLayerQuality.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayer.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRProjectionLayer::startFrame):
(WebCore::XRProjectionLayer::endFrame):
* Source/WebCore/Modules/webxr/XRProjectionLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRProjectionLayer::textureWidth const):
(WebCore::XRProjectionLayer::textureHeight const):
(WebCore::XRProjectionLayer::textureArrayLength const):
(WebCore::XRProjectionLayer::ignoreDepthValues const):
(WebCore::XRProjectionLayer::fixedFoveation const):
(WebCore::XRProjectionLayer::setFixedFoveation):
(WebCore::XRProjectionLayer::deltaPose const):
(WebCore::XRProjectionLayer::setDeltaPose):
* Source/WebCore/Modules/webxr/XRProjectionLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRProjectionLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayer.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRQuadLayer::space const):
(WebCore::XRQuadLayer::setSpace):
(WebCore::XRQuadLayer::transform const):
(WebCore::XRQuadLayer::setTransform):
(WebCore::XRQuadLayer::width const):
(WebCore::XRQuadLayer::setWidth):
(WebCore::XRQuadLayer::height const):
(WebCore::XRQuadLayer::setHeight):
* Source/WebCore/Modules/webxr/XRQuadLayer.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayerInit.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRQuadLayerInit.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRSubImage.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRSubImage::viewport const):
* Source/WebCore/Modules/webxr/XRSubImage.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRTextureType.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRTextureType.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLBinding.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRWebGLBinding::create):
* Source/WebCore/Modules/webxr/XRWebGLBinding.h: Added.
(WebCore::XRWebGLBinding::nativeProjectionScaleFactor const):
(WebCore::XRWebGLBinding::usesDepthValues const):
(WebCore::XRWebGLBinding::createProjectionLayer):
(WebCore::XRWebGLBinding::createQuadLayer):
(WebCore::XRWebGLBinding::createCylinderLayer):
(WebCore::XRWebGLBinding::createEquirectLayer):
(WebCore::XRWebGLBinding::createCubeLayer):
(WebCore::XRWebGLBinding::getSubImage):
(WebCore::XRWebGLBinding::getViewSubImage):
* Source/WebCore/Modules/webxr/XRWebGLBinding.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLSubImage.cpp: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Modules/webxr/XRWebGLSubImage.h: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
(WebCore::XRWebGLSubImage::colorTexture const):
(WebCore::XRWebGLSubImage::depthStencilTexture const):
(WebCore::XRWebGLSubImage::motionVectorTexture const):
(WebCore::XRWebGLSubImage::imageIndex const):
(WebCore::XRWebGLSubImage::colorTextureWidth const):
(WebCore::XRWebGLSubImage::colorTextureHeight const):
(WebCore::XRWebGLSubImage::depthStencilTextureWidth const):
(WebCore::XRWebGLSubImage::depthStencilTextureHeight const):
(WebCore::XRWebGLSubImage::motionVectorTextureWidth const):
(WebCore::XRWebGLSubImage::motionVectorTextureHeight const):
* Source/WebCore/Modules/webxr/XRWebGLSubImage.idl: Copied from Source/WebCore/Modules/webxr/WebXRLayer.h.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventInterfaces.in:
* Source/WebCore/dom/EventNames.json:
* Source/WebCore/html/HTMLAttributeNames.in:

Canonical link: https://commits.webkit.org/282031@main
